### DISCUSSION TITLE
up-hungry append

### DISCUSSION THREAD

**Discussion by tobyzerner** [unpoly/unpoly#305] [Q&A] [answered]
From the docs:

> Elements with an [up-hungry] attribute are updated whenever the server sends a matching element, even if the element isn't targeted.
>
> Use cases for this are unread message counters or notification flashes. Such elements often live in the layout, outside of the content area that is being replaced.

When a [up-hungry] element is matched, instead of replacing the entire element and its subtree, is it possible to append its child elements to the existing element? Similar to the effect of using the `:after` fragment pseudo-selector.

An example use-case for this is having a permanent toast notifications container. Rather than cleaning it out on each request, I would like to append new toast notifications without removing existing ones.

If it's not baked into the API, I can imagine a possible solution listening for a "before this fragment is updated" event on the up-hungry element... but I can't seem to find such an event.

**Comment by adam12** [comment, marked as answer]
It's an interesting use-case. The UX would have to be considered here, but maybe it's as simple as `up-hungry` accepting a pseudo selector.

If I was to approach your specific use-case, I think I'd have a compiler that moved the notifications for me (perhaps from one container to another). I didn't test this, but even using `up.render` might be a possibility by passing the `content` and `target` options.


```
up.compiler(".pending-notification", function(el) {
  up.render({ target: ".notifications:after", content: el, animation: "fade-in" })
})
```
